tools/mmp/driver.cs

```diff
@@ -551,9 +551,19 @@ static bool ParseBool (string value, string name)
 		}
 
 
+		const string XcodeDefault = "/Applications/Xcode.app";
 		static void ValidateXcode ()
 		{
 			if (xcode_version == null) {
+				if (sdk_root == null) {
+					if (File.Exists (Path.Combine (XcodeDefault, "Contents/MacOS/Xcode"))) {
+						sdk_root = XcodeDefault;
+						ErrorHelper.Warning (62, "No Xcode.app specified (using --sdkroot or 'xcode-select --print-path'), using the default Xcode instead: {0}", sdk_root);
```

Comment:
You're not actually checking the system here (even though the warning message says so). Can't we just move this entire chunk of code to the shared Driver.cs and fixup iOS specific stuff: https://github.com/xamarin/xamarin-macios/blob/363eb08/tools/mtouch/mtouch.cs#L319-L397?